core/src/parser.rs

```diff
@@ -618,54 +636,115 @@ fn is_redacted(attrs: &[syn::Attribute]) -> bool {
     })
 }
 
-/// Check if we have a `target_os` cfg that dooes not match command line
-/// argument `--target-os`.
+/// Check if this attribute rejects our optional `target_os` arguments.
```

Comment:
Admittedly I don't have a ton of experience with the `get_meta_items` and similar parsing here, however I find it a bit difficult to track what combinations are being checked through these methods. Could we add some comments (or add to the doc comments) describing what kind of expression each of these is parsing and how?

Related, given that `any`, `all`, `not`, etc. can be combined in basically any way, are there groups that we don't support? If so, should we document what settings do and do not work with typeshare?

Reply:
I've updated the comments for the two functions.

All the use cases are in the `core/data/tests/excluded_by_target_os/input.rs` snapshot test.